services/src/main/java/org/keycloak/userprofile/UserProfileUtils.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.userprofile;
+
+import java.util.List;
+
+import org.keycloak.events.Details;
+import org.keycloak.events.Event;
+import org.keycloak.events.EventBuilder;
+import org.keycloak.models.UserModel;
+
+/**
+ * 
+ * @author Vlastimil Elias <velias@redhat.com>
+ *
+ */
+public class UserProfileUtils {
```

Comment:
Wouldn't make more sense to create a specific listener implementation that can be re-used as this method?

Ideally, we should be handling this event within the SPI, but for now I think the approach is OK.

Reply:
Right, specific listener implementation should be better, going to implement it this way